nodejs/sortFile/utils/generateFile.js

```diff
@@ -0,0 +1,20 @@
+const fs = require("fs");
```

Comment:
здесь есть проблема в том, что мы не отслеживаем вариант, когда мы просто не можем записывать в стрим, потому что внтуренний буфер переполнен и соответственно лимит доступной памяти просто будет исчерпан, ( чтобы этого избежать можно попробовать пару вариантов, например, такой:
```js
import fs from 'node:fs';
import path from 'node:path';

const fileSize = process.argv[2];
const filePath = process.argv[3];

const writer = fs.createWriteStream(path.resolve(filePath));

function getRandomInt(min, max) {
  min = Math.ceil(min);
  max = Math.floor(max);
  return Math.floor(Math.random() * (max - min) + min);
}

for (let i = 0; i < fileSize; i++) {
  await new Promise((res, rej) => {
    const number = getRandomInt(0, fileSize);

    writer.write(`${number}\n`, (err) => {
      if (err) rej();

      res();
    });
  });
}
```
или вариант из самой документации:

```js
// Write the data to the supplied writable stream one million times.
// Be attentive to back-pressure.
function writeOneMillionTimes(writer, data, encoding, callback) {
  let i = 1000000;
  write();
  function write() {
    let ok = true;
    do {
      i--;
      if (i === 0) {
        // Last time!
        writer.write(data, encoding, callback);
      } else {
        // See if we should continue, or wait.
        // Don't pass the callback, because we're not done yet.
        ok = writer.write(data, encoding);
      }
    } while (i > 0 && ok);
    if (i > 0) {
      // Had to stop early!
      // Write some more once it drains.
      writer.once('drain', write);
    }
  }
}
```
я считаю что он лучше, просто потому что быстрее

и ещё я не уверен, что тут генерируется корректный размер файла, если я хочу сгенерировать 100мб, что мне нужно передать в качестве аргументов? 1024 * 100 * 1000? тогда мы получим 102.400.000 строк, но это не 100 мб, потому что каждая строка содержит больше чем байт (число с разрядностью больше 1го)

Reply:
да, тут генерируется не файл размером 100Mb. Практически у меня генерировался файл размером примерно 83Mb